{"content":"\r\n  \r\n  <h1>\r\n    <span class=\"lang-csharp\">WorkflowUnhandledExceptionAction</span>\r\n      Enum\r\n  </h1>\r\n  \r\n  <h2 class=\"hiddenAnchor\">Definition</h2>\r\n  \r\n  <div class=\"metadata\">\r\n  \r\n  \r\n  </div>\r\n  \r\n  <div class=\"summaryHolder\">\r\n  \t<div class=\"summary\">\r\n  \t\t<p>An enumeration that specifies the action to perform when an unhandled exception occurs within a workflow.</p>\n\r\n  \t</div>\r\n  </div><nav id=\"center-doc-outline\" class=\"doc-outline\" ms.cmpgrp=\"intopic toc\" role=\"navigation\" aria-label=\"On page navigation\">\r\n    <h3>In this Article</h3>\r\n  </nav>\r\n  \t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\">public enum WorkflowUnhandledExceptionAction</code></pre>\r\n  \r\n  \r\n  \r\n  \r\n  \t<dl class=\"attributeList stack\">\r\n  \t\t<dt>Inheritance</dt><dd>\r\n  \t\t<div class=\"pdlHolder\">\r\n  \t\t<div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object\" data-linktype=\"relative-path\">Object</a></span></div><div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.valuetype\" data-linktype=\"relative-path\">ValueType</a></span></div><div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.enum\" data-linktype=\"relative-path\">Enum</a></span></div><span class=\"lang-csharp\">WorkflowUnhandledExceptionAction</span>\r\n  \t\t</div>\r\n  \t\t</dd>\r\n  \t</dl>\r\n  \r\n  \r\n  \r\n  \r\n  \r\n  \r\n  \r\n      \t<h2 class=\"accented\">Fields\r\n      </h2>\r\n      <table class=\"nameValue\">\r\n      \t<tr><td id=\"System_ServiceModel_Activities_Description_WorkflowUnhandledExceptionAction_Abandon\">\r\n      \t\t\t<span class=\"lang-csharp\">Abandon</span>\r\n      \t</td>\r\n      \t<td><p>Abandons the workflow instance.</p>\n</td>\r\n      \t</tr>\r\n      \t<tr><td id=\"System_ServiceModel_Activities_Description_WorkflowUnhandledExceptionAction_AbandonAndSuspend\">\r\n      \t\t\t<span class=\"lang-csharp\">AbandonAndSuspend</span>\r\n      \t</td>\r\n      \t<td><p>Abandons the workflow instance and marks the last persisted state as suspended. If there is no last persisted instance can be found, the workflow instance is aborted.</p>\n</td>\r\n      \t</tr>\r\n      \t<tr><td id=\"System_ServiceModel_Activities_Description_WorkflowUnhandledExceptionAction_Cancel\">\r\n      \t\t\t<span class=\"lang-csharp\">Cancel</span>\r\n      \t</td>\r\n      \t<td><p>Cancels the workflow instance.</p>\n</td>\r\n      \t</tr>\r\n      \t<tr><td id=\"System_ServiceModel_Activities_Description_WorkflowUnhandledExceptionAction_Terminate\">\r\n      \t\t\t<span class=\"lang-csharp\">Terminate</span>\r\n      \t</td>\r\n      \t<td><p>Terminates the workflow instance. The instance state is <a class=\"xref\" href=\"system.activities.hosting.workflowinstancestate\" data-linktype=\"relative-path\">WorkflowInstanceState</a>.</p>\n</td>\r\n      \t</tr>\r\n      </table>\r\n  \r\n  \r\n","outputRootRelativePath":"../","pageMetadata":"<meta name=\"author\" content=\"dotnet-bot\">\r\n<meta name=\"ms.author\" content=\"dotnetcontent\">\r\n<meta name=\"manager\" content=\"wpickett\">\r\n<meta name=\"breadcrumb_path\" content=\"../toc2.json\">\r\n<meta name=\"apiPlatform\" content=\"dotnet\">\r\n<meta name=\"ms.topic\" content=\"managed-reference\">\r\n<meta name=\"ms.prod\" content=\".net\">\r\n<meta name=\"uhfHeaderId\" content=\"MSDocsHeader-DotNet\">\r\n<meta name=\"search.ms_sitename\" content=\"Docs\">\r\n<meta name=\"search.ms_docsetname\" content=\"docs-internal\">\r\n<meta name=\"version\" content=\"0\">\r\n<meta name=\"locale\" content=\"en-us\">\r\n<meta name=\"site_name\" content=\"Docs\">\r\n<meta name=\"search.ms_product\" content=\"MSDN\">\r\n<meta name=\"depot_name\" content=\"MSDN.docs-internal\">\r\n<meta name=\"original_ref_skeleton_git_url\" content=\"https://github.com/dotnet/docs-internal/blob/master/api/System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.yml\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Abandon\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.AbandonAndSuspend\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Cancel\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Terminate\">\r\n<meta name=\"APILocation\" content=\"System.ServiceModel.Activities.dll\">\r\n<meta name=\"TopicType\" content=\"apiref\">\r\n<meta name=\"APIType\" content=\"Assembly\">\r\n<meta name=\"updated_at\" content=\"2017-04-01 05:59 PM\">\r\n<meta name=\"document_id\" content=\"f3d50030-65f8-2743-9162-f70ad53192a5\">\r\n<meta name=\"pagetype\" content=\"Reference\">\r\n<meta name=\"description\" content=\"An enumeration that specifies the action to perform when an unhandled exception occurs within a workflow.\n\">\r\n<meta name=\"toc_rel\" content=\"_splitted/system.servicemodel.activities.description/toc2.json\">\r\n<meta name=\"source_url\" content=\"\">\r\n<meta name=\"ms.assetid\" content=\"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction\">\r\n","rawMetadata":{"author":"dotnet-bot","ms.author":"dotnetcontent","manager":"wpickett","breadcrumb_path":"/dotnet/toc.json","apiPlatform":"dotnet","ms.topic":"managed-reference","ms.prod":".net","uhfHeaderId":"MSDocsHeader-DotNet","search.ms_sitename":"Docs","search.ms_docsetname":"docs-internal","version":0,"_op_canonicalUrlPrefix":"https://docs.microsoft.com/en-us/dotnet-internal/","locale":"en-us","site_name":"Docs","search.ms_product":"MSDN","_op_openToPublicContributors":false,"depot_name":"MSDN.docs-internal","_op_gitRefSkeletonCommitHistory":[],"original_ref_skeleton_git_url":"https://github.com/dotnet/docs-internal/blob/master/api/System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.yml","open_to_public_contributors":false,"api_name":["System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Abandon","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.AbandonAndSuspend","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Cancel","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Terminate"],"api_location":["System.ServiceModel.Activities.dll"],"topic_type":["apiref"],"api_type":["Assembly"],"f1_keywords":["System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction","System::ServiceModel::Activities::Description::WorkflowUnhandledExceptionAction","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Abandon","System::ServiceModel::Activities::Description::WorkflowUnhandledExceptionAction::Abandon","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.AbandonAndSuspend","System::ServiceModel::Activities::Description::WorkflowUnhandledExceptionAction::AbandonAndSuspend","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Cancel","System::ServiceModel::Activities::Description::WorkflowUnhandledExceptionAction::Cancel","System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.Terminate","System::ServiceModel::Activities::Description::WorkflowUnhandledExceptionAction::Terminate"],"dev_langs":["csharp"],"updated_at":"2017-04-01 05:59 PM","document_id":"f3d50030-65f8-2743-9162-f70ad53192a5","content_git_url":"https://github.com/dotnet/docs-internal/blob/master/xml/System.ServiceModel.Activities.Description/WorkflowUnhandledExceptionAction.xml","layout":"Reference","_op_layout":"Reference","pagetype":"Reference","title":"WorkflowUnhandledExceptionAction enum | Microsoft Docs","_op_ogTitle":"WorkflowUnhandledExceptionAction enum","description":"An enumeration that specifies the action to perform when an unhandled exception occurs within a workflow.\n","toc_asset_id":"api/_splitted/System.ServiceModel.Activities.Description/toc.json","toc_rel":"_splitted/System.ServiceModel.Activities.Description/toc.json","source_url":"","ms.assetid":"System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction","canonical_url":"https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.activities.description.workflowunhandledexceptionaction","_op_searchScope":[".NET"],"_op_canonicalUrl":"https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.activities.description.workflowunhandledexceptionaction","fileRelativePath":"api/System.ServiceModel.Activities.Description.WorkflowUnhandledExceptionAction.html"},"themesRelativePathToOutputRoot":"_themes/"}